version: '3.5'
services:
  database-service:
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U cake -d cake || exit 1"]
      interval: 10s
      timeout: 5s
      retries: 3
    image: postgres:10
    container_name: cake-database
    environment:
    - POSTGRES_PASSWORD=senhacake
    - POSTGRES_USER=cake
    - POSTGRES_DB=cake
    - POSTGRES_DB_TEST=cake-test
    ports:
    - "5433:5432"
    expose:
    - "5432"
    volumes:
    - ./database/docker-data/postgres:/var/lib/postgresql/data
    - ./database/:/docker-entrypoint-initdb.d/
    networks:
    - cake_network
  webapp-service:
    #    build: ./webapp
    build: webapp/.
    image: node:10
    container_name: cake-web
    #user: "node"
    user: "root"
    working_dir: /home/node/app
    #    healthcheck:
    #      test: ["CMD", "curl", "-sS", "http://localhost:8080"]
    #      #      test: ["CMD-SHELL", "node /home/node/app/healthcheck.js"]
    #      interval: 20s
    #      timeout: 5s
    #      retries: 3
    volumes:
    - ./webapp:/home/node/app
    command:
      #      sh -c 'npm i && npm run serve'
      #sh -c 'npm run serve'
      sh -c 'npm i && npm run build'
    #      sh -c 'npm run watch'
    ports:
    - "8080:8080"
    environment:
    - NODE_ENV=development
    networks:
    - cake_network


      ports:
        - "8090:80"
      volumes:
        - ~/my/local/path/html:/var/www/html
      external_links:
        - docker-mysql:db
      network_mode: bridge
# docker network create cake_network

networks:
  cake_network:
    name: cake_network
    driver: bridge
